examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -173,49 +185,55 @@ void RegularGridDecomposition::exchangeHaloParticles(AutoPasType &autoPasContain
 
-    const double leftHaloMin = _localBoxMin[dimensionIndex] - _skinWidthPerTimestep * _rebuildFrequency;
-    const double leftHaloMax = _localBoxMin[dimensionIndex] + _cutoffWidth + _skinWidthPerTimestep * _rebuildFrequency;
-    const double rightHaloMin = _localBoxMax[dimensionIndex] - _cutoffWidth - _skinWidthPerTimestep * _rebuildFrequency;
-    const double rightHaloMax = _localBoxMax[dimensionIndex] + _skinWidthPerTimestep * _rebuildFrequency;
+    const double leftHaloMin = _localBoxMin[dimensionIndex] - skinWidth;
+    const double leftHaloMax = _localBoxMin[dimensionIndex] + _cutoffWidth + skinWidth;
+    const double rightHaloMin = _localBoxMax[dimensionIndex] - _cutoffWidth - skinWidth;
+    const double rightHaloMax = _localBoxMax[dimensionIndex] + skinWidth;
 
-    for (const auto &particle : haloParticles) {
+    for (const auto &particle : _haloParticles) {
       std::array<double, _dimensionCount> position = particle.getR();
       if (position[dimensionIndex] >= leftHaloMin and position[dimensionIndex] < leftHaloMax) {
-        particlesForLeftNeighbor.push_back(particle);
+        _particlesForLeftNeighbor.push_back(particle);
 
         // Apply boundary condition
         if (isNear(_localBoxMin[dimensionIndex], _globalBoxMin[dimensionIndex])) {
           position[dimensionIndex] =
               position[dimensionIndex] + (_globalBoxMax[dimensionIndex] - _globalBoxMin[dimensionIndex]);
-          particlesForLeftNeighbor.back().setR(position);
+          _particlesForLeftNeighbor.back().setR(position);
         }
       } else if (position[dimensionIndex] >= rightHaloMin and position[dimensionIndex] < rightHaloMax) {
-        particlesForRightNeighbor.push_back(particle);
+        _particlesForRightNeighbor.push_back(particle);
 
         // Apply boundary condition
         if (isNear(_localBoxMax[dimensionIndex], _globalBoxMax[dimensionIndex])) {
           position[dimensionIndex] =
               position[dimensionIndex] - (_globalBoxMax[dimensionIndex] - _globalBoxMin[dimensionIndex]);
-          particlesForRightNeighbor.back().setR(position);
+          _particlesForRightNeighbor.back().setR(position);
         }
       }
     }
     // See documentation for _neighborDomainIndices to explain the indexing
     const int leftNeighbor = _neighborDomainIndices[(dimensionIndex * 2) % _neighborCount];
     const int rightNeighbor = _neighborDomainIndices[(dimensionIndex * 2 + 1) % _neighborCount];
-    const auto receivedHaloParticles = sendAndReceiveParticlesLeftAndRight(
-        particlesForLeftNeighbor, particlesForRightNeighbor, leftNeighbor, rightNeighbor);
-    haloParticles.insert(haloParticles.end(), receivedHaloParticles.begin(), receivedHaloParticles.end());
+    // clear the buffer as it will hold the newly received halo particles.
+    _receivedParticlesBuffer.clear();
+    sendAndReceiveParticlesLeftAndRight(_particlesForLeftNeighbor, _particlesForRightNeighbor, _receivedParticlesBuffer,
```

Comment:
```suggestion
    sendAndReceiveParticlesLeftAndRight(_particlesForLeftNeighbor, _particlesForRightNeighbor, _haloParticles,
```

If my comment above is correct, then we could directly receive in `_haloParticles`, since it should be empty here.

Reply:
In theory, we could receive directly into `_haloParticles`, but this would require rewriting a bunch of send / receive functions so that we can define where in the vector we want to store the new stuff so that we can append the new particles. But since this doesn't (yet?) show up in the profilers, I can't be bothered to make the changes :stuck_out_tongue_winking_eye: